crates/core/tedge_agent/src/operation_file_cache/mod.rs

```diff
@@ -0,0 +1,364 @@
+            }
+        }
+
+        Ok(())
+    }
+}
+
+impl FileCacheActor {
+    async fn process_mqtt_message(
+        &mut self,
+        mqtt_message: MqttMessage,
+    ) -> Result<(), RuntimeError> {
+        let Ok((entity, channel)) = self.mqtt_schema.entity_channel_of(&mqtt_message.topic) else {
+            return Ok(());
+        };
+
+        let Channel::Command {
+            operation: OperationType::ConfigUpdate,
+            cmd_id,
+        } = channel
+        else {
+            return Ok(());
+        };
+
+        let update_payload =
+            match serde_json::from_slice::<ConfigUpdateCmdPayload>(mqtt_message.payload.as_bytes())
+            {
+                Ok(payload) => payload,
+                Err(err) => {
+                    warn!("Received config update, but payload is malformed: {err}");
+                    return Ok(());
+                }
+            };
+
+        if update_payload.remote_url.is_empty() || update_payload.tedge_url.is_some() {
+            return Ok(());
+        }
+
+        match &update_payload.status {
+            CommandStatus::Executing => {
+                self.download_config_file_to_cache(&mqtt_message.topic, &update_payload)
+                    .await?;
+            }
+            CommandStatus::Successful => self.delete_symlink_for_config_update(
+                &entity,
+                &update_payload.config_type,
+                &cmd_id,
+            )?,
+            CommandStatus::Failed { .. } => self.delete_symlink_for_config_update(
+                &entity,
+                &update_payload.config_type,
+                &cmd_id,
+            )?,
+            _ => {}
+        }
+
+        Ok(())
+    }
+
+    async fn process_download(&mut self, download: IdDownloadResult) -> Result<(), RuntimeError> {
+        let (topic, result) = download;
+
+        let Some(mut operation) = self.pending_operations.remove(&topic) else {
+            return Ok(());
+        };
+
+        let (entity, Channel::Command { cmd_id, .. }) = self
+            .mqtt_schema
+            .entity_channel_of(&Topic::new(&topic).unwrap())
+            .expect("only topics targetting config update command should be inserted")
+        else {
+            return Ok(());
+        };
+
+        let download = match result {
+            // if cant download file, operation failed
+            Err(err) => {
+                let error_message = format!("tedge-agent failed downloading a file: {err}");
+                operation.failed(&error_message);
+                error!("{}", error_message);
+                let message = MqttMessage::new(
+                    &Topic::new(&topic).unwrap(),
+                    serde_json::to_string(&operation).unwrap(),
+                );
+                self.mqtt_sender.send(message).await?;
+                return Ok(());
+            }
+            Ok(download) => download,
+        };
+
+        self.create_symlink_for_config_update(
```

Comment:
What's the purpose of this symlink?

Reply:
I'm not sure if I understand the question. It should be the same as before: we download the file to the data dir cache directory, and then create the symlink to this file in the file transfer directory. After the operation is complete, we delete the symlink.

Reply:
Ok thanks. I though this have been introduced by this PR.

@rina23q do you know the answer?

Reply:
The file-transfer service only allows you to share files under the `/var/tedge/file-transfer` directory (or at least that was the case earlier). Since the cached files are downloaded to a different location (`/var/tedge/cache`) which doesn't fall under the `file-transfer` dir, we just create symlinks under that directory pointing to the cached file.

Reply:
What @Bravo555 said is totally correct.

Creating a symlink from cache is more about the security. Since the files under `/var/tedge/file-transfer` can be accessible from all other thin-edge components, that means they can modify the contents by using HTTP requests. Keeping `/var/tedge/cache` out of the file transfer directory ensures the files cannot change by any HTTP requests.

Reply:
> Keeping /var/tedge/cache out of the file transfer directory ensures the files cannot change by any HTTP requests.

Good point @rina23q